tests/integration/integration-tests.sh

```diff
@@ -117,29 +125,41 @@ setup_test_pod() {
     fi
     SLEEP_TIME=$(expr $SLEEP_TIME \* 2);
   done
+}
+
+setup_admin_test_pod() {
+  kubectl create ns $NAMESPACE
+
+  echo "Creating admin test-pod"
+
+  # Create a pod with admin service account
+  yq ea '.spec.containers[0].env[0].name = "KUBECONFIG" | .spec.containers[0].env[0].value = "/var/lib/spark/.kube/config" | .metadata.name = "testpod-admin"' \
+    ./tests/integration/resources/testpod.yaml | \
+    kubectl -n tests apply -f -
+
+  wait_for_pod testpod-admin $NAMESPACE
 
   MY_KUBE_CONFIG=$(cat /home/${USER}/.kube/config)
-  TEST_POD_TEMPLATE=$(cat tests/integration/resources/podTemplate.yaml)
 
-  kubectl exec testpod -- /bin/bash -c 'mkdir -p ~/.kube'
-  kubectl exec testpod -- env KCONFIG="$MY_KUBE_CONFIG" /bin/bash -c 'echo "$KCONFIG" > ~/.kube/config'
-  kubectl exec testpod -- /bin/bash -c 'cat ~/.kube/config'
-  kubectl exec testpod -- /bin/bash -c 'cp -r /opt/spark/python /var/lib/spark/'
-  kubectl exec testpod -- env PTEMPLATE="$TEST_POD_TEMPLATE" /bin/bash -c 'echo "$PTEMPLATE" > /etc/spark/conf/podTemplate.yaml'
+  kubectl -n $NAMESPACE exec testpod-admin -- /bin/bash -c 'mkdir -p ~/.kube'
+  kubectl -n $NAMESPACE exec testpod-admin -- env KCONFIG="$MY_KUBE_CONFIG" /bin/bash -c 'echo "$KCONFIG" > ~/.kube/config'
 }
 
 teardown_test_pod() {
-  kubectl delete pod testpod
+  kubectl -n $NAMESPACE delete pod testpod
+  kubectl -n $NAMESPACE delete pod testpod-admin
+
+  kubectl delete namespace $NAMESPACE
```

Comment:
Yes, that would be the same on the happy path. But I noticed that in general we were removing single pods before the namespace and I did not mind (since these are tests) to have similar approach and to be declarative, effectively having an extra check for free (if the deletion of pods fails because the pod does not exists)